### ISSUE TITLE
Lost in examples

### ISSUE THREAD

**Issue by Seraf** [noirbizarre/flask-restplus#38]
Hello,

First, thanks for your very great work !
I'm a little lost with examples, I would like to create a complete CRUD with a simple model named Role which have 2 fields : "name" and "description".

Here is the complete file : https://gist.github.com/Seraf/23d0d34e11c38f922f4b

I think most of the docs/api is correct, but I guess I have a misunderstanding about the parser and how to use it. Sometimes examples use parsers, sometimes the class fields.Raw (Person) ...

Can you give me some advice/comment about it please ?

Thanks

**Comment by noirbizarre** [comment]
Hello,

Thanks for the compliment.

The parser is simply a flask-restful parser.
It's mainly used to extract simple attributes from the query strings, the headers or the request payload (form, file or json).

Models (and fields) purpose is to describe complex structure extracted from the request payload (it allows nested, list...).

You can mix them depending on your needs.

My personal use case, is to use the request parser to validate query parameters, headers and file upload and to use models coupled with ``@api.marshal_with()``, ``@api.expect()`` and ``@api.response()`` to describe payloads (expected and returned).

My APIs expect and respond JSON in their payloads (with the file upload/download exceptions), but given your gist, you seems to use formdata.
I forked and modified your gist: https://gist.github.com/noirbizarre/fe30e4578f312853638f

- only specify a model when you return (40X responses does not return the expected model)
- ``@api.marshal_with`` decorator document your response, so only use ``@api.response`` for other cases (errors)
- try to return the object and a status code (if different from the default 200) instead of the verbatim for the status code

If you want to be more restful (I know, this is a buzz word), try to use JSON as input instead of formdata.

Payload validation for model will come in next release with [jsonschema](https://pypi.python.org/pypi/jsonschema).

**Comment by Seraf** [comment]
Hello,

Thank you very much for taking the time to modify my gist. It seems more clear now, thank for the explanation. I will try to use JSON instead of formdata :)

**Comment by Seraf** [comment]
I tested your revision of my file and everything works very well except this little thing : 
@api.marshal_list_with(role_api, description="Roles list")
TypeError: marshal_list_with() got an unexpected keyword argument 'description'

Any idea about this please ?
Flask-RESTful==0.3.2
flask-restplus==0.7.0

**Comment by noirbizarre** [comment]
Sorry, I fixed this in the 0.7.1.
You can update and retry.

**Comment by Seraf** [comment]
Thanks it works great !

**Comment by Seraf** [comment]
@noirbizarre me again for new question

I decided to use name as argument to find my role.
I updated it like this : https://gist.github.com/Seraf/23d0d34e11c38f922f4b

The get(name) and delete(name) works great. But the PUT method has a problem : 
http://127.0.0.1:5000/api/1/core/role/{name} -> the name here isn't replaced by the name value I give
On the get function, it uses the name I give : 
http://127.0.0.1:5000/api/1/core/role/test

I think the problem is that PUT method use the name of my model and not the param field in my decorator.
I think I can workaround the problem using "rolename" as param but it won't make sense on my PUT method to have rolename param and name param with the same value.
I hope I'm clear in my explanation :)

PS : the goal of this mini project is to provide a flask API with plugin support (using stevedore) to let user create plugins and automatically expose it in the API (and generate documentation with swagger). So I'm actually focusing on the user/role management which is a good usecase to use sqlalchemy models with nested fields etc.
